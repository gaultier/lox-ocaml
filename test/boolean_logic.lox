print ((1 != 3) == !!-2); // expect: true

// or
print "";  // expect:  
print 1 or 2;// expect: 1.

print 1 or nil; // expect: 1. 

print nil or 5; // expect: 5.
print nil or "hey"; // expect: hey

print nil or false; // expect: false

print nil or nil; // expect: nil

print "joe" or "jane"; // expect: joe

print nil or true; // expect: true

print "bill" or true; // expect: bill

// and
print ""; // expect:  
print 1 and 2; // expect: 2.

print 1 and nil; // expect: nil

print nil and 5; // expect: nil

print nil and "hey"; // expect: nil

print nil and false; // expect: nil

print nil and nil; // expect: nil

print "joe" and "jane"; // expect: jane

print nil and true; // expect: nil

print "bill" and true; // expect: true

// mixed
print ""; // expect:  
print 1 or 2 or 3; // expect: 1.

print 1 and (3 or 2); // expect: 3.

print !(1 < 2) or ("hello" and "world");  // expect: world
