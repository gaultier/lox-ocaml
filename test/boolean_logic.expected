true
1.
1. 
5.
hey
false
nil
joe
true
bill
2.
nil
nil
nil
nil
nil
jane
nil
true
1.
3.
world
